onnx/defs/generator/defs.cc

```diff
@@ -38,10 +70,26 @@ ONNX_OPERATOR_SET_SCHEMA(
         .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
           auto* value = ctx.getAttribute("value");
           auto* sparse_value = ctx.getAttribute("sparse_value");
-
-          if ((nullptr != value) && (nullptr != sparse_value))
+          auto* value_int = ctx.getAttribute("value_int");
+          auto* value_ints = ctx.getAttribute("value_ints");
+          auto* value_float = ctx.getAttribute("value_float");
+          auto* value_floats = ctx.getAttribute("value_floats");
+          auto* value_string = ctx.getAttribute("value_string");
+          auto* value_strings = ctx.getAttribute("value_strings");
+
+          std::vector<int> non_null_attr = {
+            (nullptr != value),
+            (nullptr != sparse_value),
+            (nullptr != value_int),
+            (nullptr != value_ints),
+            (nullptr != value_float),
+            (nullptr != value_floats),
+            (nullptr != value_string),
+            (nullptr != value_strings)
+          };
+          if (std::count(non_null_attr.begin(), non_null_attr.end(), true) != 1)
             fail_shape_inference(
-                "Only one of the attributes 'value' or 'sparse_value' must be specified for a Constant node.");
+                "One and only one of the attributes 'value', 'value_*' or 'sparse_value' must be specified for a Constant node.");
```

Comment:
The constraint changed: Here at least one attribute should be defined AND no more than one attribute should be defined.

Reply:
There's also a message on line 168 that needs updating, although it should never be reached now that you check the `<= 1` part at the top.